### PR TITLE
Feature: New file explorer

### LINKED ISSUE #614: Feature: File explorer
We should have a file explorer where folder and files are listed and clicking on a folder would show the folder content.

### PR DESCRIPTION
**Pull request by timotheeguerin**
fix #614 

This remove the file page as all the file preview are now inline. This should reduce the number of clicks

image
image

Task output new design
image

image

### CODE REVIEW

File: app/components/file/browse/file-explorer/file-tree-view/file-tree-view.component.ts [hunk 1]
```diff
@@ -0,0 +1,170 @@
...
+    @Input() public autoExpand = false;
+    @Output() public navigate = new EventEmitter<string>();
+
+    public expanded = true;
+    public currentNode: FileTreeNode;
+    public expandedDirs: StringMap<boolean> = {};
+
+    public treeRows: TreeRow[] = [];
+    public refreshing: boolean;
+    private _tree: FileTreeStructure;
```
Comment (ascobie) [inline app/components/file/browse/file-explorer/file-tree-view/file-tree-view.component.ts]: extreme nitpick, space between public and private. :)

File: app/components/job/job-hook-task/job-hook-task-details/job-hook-task-details.html [hunk 2]
```diff
@@ -1,9 +1,10 @@
-    <div *ngIf="task">
+<div *ngIf="task">
     <bl-property-group>
         <bl-link-property label="Pool" [value]="task.poolId" [link]="['/pools', task.poolId]"></bl-link-property>
         <bl-link-property label="Node" [value]="task.nodeId" [link]="['/pools', task.poolId, 'nodes', task.nodeId]"></bl-link-property>
-
-        <bl-node-file-list #list [poolId]="task.poolId" [nodeId]="task.nodeId" [folder]="currentFolder" *ngIf="currentFolder">
-        </bl-node-file-list>
+        Cur: {{currentFolder}}
```
Comment (ascobie) [inline app/components/job/job-hook-task/job-hook-task-details/job-hook-task-details.html]: should this be "Current"? or is "Cur" ok here?

File: app/services/file/file-navigator/file-tree.model.ts [hunk 3]
```diff
@@ -0,0 +1,143 @@
...
+        this.contentLength = params.contentLength;
+        this.lastModified = params.lastModified;
+
+        this.name = this._computeName();
+    }
+
+    public walk() {
+        return this.children.values();
+    }
+    public markAsLoaded() {
```
Comment (ascobie) [inline app/services/file/file-navigator/file-tree.model.ts]: nitpick: newline between methods.

File: app/services/file/file-navigator/file-tree.model.ts [hunk 4]
```diff
@@ -0,0 +1,143 @@
...
+
+    public getNode(nodePath: string) {
+        nodePath = CloudPathUtils.normalize(nodePath);
+        if (nodePath === "") { nodePath = "."; }
+        if (nodePath in this.directories) {
+            return this.directories[nodePath];
+        } else {
+            const parent = path.dirname(nodePath);
+            if (parent in this.directories) {
+                // const matchingChild = this.directories[parent].children.filter(x => x.path === nodePath).first();
```
Comment (ascobie) [inline app/services/file/file-navigator/file-tree.model.ts]: remove if not needed.

File: app/services/file/file-navigator/helper.ts [hunk 5]
```diff
@@ -0,0 +1,94 @@
...
+
+/**
+ * Helper function that map File to tree nodes. It will build the nested data structure
+ */
+export function mapFilesToTree(files: List<File>, baseFolder: string = ""): FileTreeNode[] {
+    const directories = {};
+
+    for (let file of files.toArray()) {
+        const node = fileToTreeNode(file);
+
```
Comment (ascobie) [inline app/services/file/file-navigator/helper.ts]: probably remove these newlines to be consistent.

File: app/components/job/job-hook-task/job-hook-task-details/job-hook-task-details.html (hunk 2 above)
Comment (timotheeguerin) [reply app/components/job/job-hook-task/job-hook-task-details/job-hook-task-details.html]: nope should not be there, debugging, good catch

**Pull request merged**